src/mqt/predictor/rl/cost_model.py

```
@@ -0,0 +1,280 @@
+
+    # Determine the basis gates for this device
+    basis_gates = KNOWN_DEVICE_BASES.get(device_id)
+    if basis_gates is None:
+        warnings.warn(
+            f"No native gate-set defined for device '{device_id}'. "
+            "Generating cost table using a minimal universal basis (Qiskit default). "
+            "Results may be inaccurate. Consider specifying the gate set in KNOWN_DEVICE_BASES.",
+            UserWarning,
+            stacklevel=2,
+        )
+        logger.warning(f"No basis for device '{device_id}', using minimal universal basis for cost generation.")
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)